docs/dts-tests.md

```diff
@@ -0,0 +1,44 @@
+# DTS tests
+
+This document describes current DTS tests coverage. Tests planned:
+
+* E2E tests;
+* Unit tests.
+
+## E2E tests
+
+E2E - End to End tests, tests that verify how DTS goes from start to end for
+every user workflow (e.g. installation, update, etc.) for every platform.
+
+Location in OSFV: `dts/dts-e2e.robot`.
+
+These tests include modifications for DTS platform emulation so the tests
+could be launched on Qemu. Then every test case choose the workflow by
+choosing DTS menu option, and provides credentials (if necessary). So, the
+start conditions are: platform configuration and workflow selection.
+
+Then every test case goes through the chosen workflow and checks for expected
+behavior. If everything goes as expected - test case passes, otherwise - it
+fails.
+
+Control variables:
+
+* `dts_ipxe_link`: useful if you are testing DTS which is not released yet. Just
+  put here a link to your script which will load your DTS. By default DTS is
+  being booted from `dl.3mdeb.com`;
+* `dpp_logs_key`, `dpp_download_key`, `dpp_password`: for DPP credentials, if
+  tests need them.
+
+Launching example:
+
+```bash
+robot -b command_log.txt -v snipeit:no -L TRACE -v config:qemu -v rte_ip:127.0.0.1 -v netboot_utilities_support:True -v dts_ipxe_link:http://192.168.0.102:8080/ipxe -v dpp_logs_key:'LOGS_KEY' -v dpp_download_key:'DOWNLOAD_KEY' -v dpp_password:'PASSWORD' -t "E2E006.002*" dts/dts-e2e.robot
```

Comment:
`-v netboot_utilities_support:True` - why is this needed?

Reply:
This suite https://github.com/Dasharo/open-source-firmware-validation/blob/develop/dasharo-compatibility/network-boot-utilities.robot is specific to Protectli platforms

Reply:
https://github.com/Dasharo/open-source-firmware-validation/blob/b40b2f273b8866917398aef8ccbf1b990717724a/lib/dts-lib.robot#L51
From what I understand so we can boot custom DTS via iPXE shell

Reply:
```
    ELSE IF    '${dts_booting_method}'=='iPXE'
        IF    ${NETBOOT_UTILITIES_SUPPORT} == ${TRUE}
            # DTS_IPXE_LINK can be defined before running tests, e.g. via CMD or
            # some file:
            Boot Dasharo Tools Suite Via IPXE Shell    ${DTS_IPXE_LINK}
        ELSE
            ${boot_menu}=    Enter Boot Menu Tianocore And Return Construction
            Enter Submenu From Snapshot    ${boot_menu}    ${IPXE_BOOT_ENTRY}
            ${ipxe_menu}=    Get IPXE Boot Menu Construction
            Enter Submenu From Snapshot    ${ipxe_menu}    Dasharo Tools Suite
            Set DUT Response Timeout    5m
            Read From Terminal Until    .cpio.gz...
            Read From Terminal Until    ok
        END
    ELSE
        FAIL    Unknown or improper connection method: ${dts_booting_method}
    END
```

The `NETBOOT_UTILITIES_SUPPORT` variable is set for Protectli devices only, as they have a bit different iPXE menu. According to this code, the `ELSE` branch is supposed to be used for boards other than `Protectli` (with regular Dasharo iPXE menu).

If the first `IF` branch works for both `Protectli` and `Dasharo` iPXE menu (on QEMU we have Dasharo menu), then maybe we do not need this if at all, and we can always use the code from `IF` branch.

Reply:
TLDR: if the current code works for you, you can remove this `IF statement` to always select first branch, and not set the NETBOOT_UTILITIES_SUPPORT variable.

Reply:
Both work on qemu, but the bootflow after `ELSE` boots DTS from boot.dasharo.com using menu options, when the bootflow after `IF` uses iPXE shell, that  enables booting DTS from custom link.

I gues I should rename `NETBOOT_UTILITIES_SUPPORT` variable then.

Reply:
Changed here cc392489aae15f9a1b8764a205ae48cce4c22839.

Reply:
Ok I get it why we need to boot from shell now. Let's stay with this approach for now.